server/src/test/java/io/crate/analyze/parser/TypeSignatureParserTest.java

```diff
@@ -47,7 +47,13 @@ public class TypeSignatureParserTest extends ESTestCase {
     @Test
     public void testParsingOfPrimitiveDataTypes() {
         for (var type : DataTypes.PRIMITIVE_TYPES) {
-            assertThat(TypeSignature.parse(type.getName())).isEqualTo(type.getTypeSignature());
+            if (type.id() == DataTypes.CHARACTER.id()) {
+                // Parametrized types with fixed length are registered with default length.
+                // bit and float_vector and not primitives, handle only char.
+                assertThat(TypeSignature.parse("character(1)")).isEqualTo(type.getTypeSignature());
```

Comment:
To elaborate more on this:

`TypeSignature.parse("character")` is not equal to `DataTypes.CHARACTER.getTypeSignature`.

Result of `TypeSignature.parse("character")` doesn't have any params - it's an empty list. `TypeSignaturesASTVisitor.getParameters` gets null context and returns an empty list.